src/EncompassRest/ResourceLocks/ResourceLocks.cs

```diff
@@ -0,0 +1,132 @@
+﻿using EncompassRest.Loans;
+using EncompassRest.Loans.Enums;
+using EncompassRest.Utilities;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace EncompassRest.ResourceLocks
+{
+    internal sealed class ResourceLocks : ApiObject
+    {
+        internal ResourceLocks(EncompassRestClient client) : 
+            base(client, "encompass/v1/resourceLocks")
+        {
+        }
+
+        public Task<ResourceLock> GetResourceLockAsync(ResourceLock resourceLock, CancellationToken cancellationToken = default)
+        {
+            Preconditions.NotNull(resourceLock, nameof(resourceLock));
+            return GetResourceLockAsync(resourceLock.Id, resourceLock.Resource.EntityId, resourceLock.Resource.EntityType, cancellationToken);
+        }
+        public Task<ResourceLock> GetResourceLockAsync(string lockId, string resourceId, EntityType resourceType, CancellationToken cancellationToken = default) => GetResourceLockAsync(lockId, resourceId, resourceType.ToString(), cancellationToken);
+        public async Task<ResourceLock> GetResourceLockAsync(string lockId, string resourceId, string resourceType, CancellationToken cancellationToken = default)
+        {
+            Preconditions.NotNullOrEmpty(resourceId, nameof(resourceId));
+            Preconditions.NotNullOrEmpty(resourceType, nameof(resourceType));
+            Preconditions.NotNullOrEmpty(lockId, nameof(lockId));
+
+            var queryParameters = new QueryParameters();
+            queryParameters.Add(new QueryParameter("resourceType", resourceType));
+            queryParameters.Add(new QueryParameter("resourceId", resourceId));
+
+            return await GetAsync<ResourceLock>(lockId, queryParameters.ToString(), nameof(GetResourceLockAsync), lockId, cancellationToken);
+        }
+
+        public Task<string> GetResourceLockRawAsync(string lockId, string queryString, CancellationToken cancellationToken = default)
+        {
+            Preconditions.NotNullOrEmpty(queryString, nameof(queryString));
+            Preconditions.NotNullOrEmpty(lockId, nameof(lockId));
+
+            return GetRawAsync(lockId, queryString, nameof(GetResourceLockRawAsync), lockId, cancellationToken);
+        }
+
+        public Task<List<ResourceLock>> GetResourceLocksAsync(string resourceId, EntityType resourceType, CancellationToken cancellationToken = default) => GetResourceLocksAsync(resourceId, resourceType.ToString(), cancellationToken);
+        public async Task<List<ResourceLock>> GetResourceLocksAsync(string resourceId, string resourceType,  CancellationToken cancellationToken = default)
```

Comment:
Remove async keyword. There's an extra space between the resourceType and cancellationToken.